internal/fs/buffer/writebuffer.go

```diff
@@ -0,0 +1,18 @@
+package buffer
+
+type WriteBuffer interface {
+	// Create creates a buffer of 2*size passed as a parameter.
+	Create(size int) WriteBuffer
+
+	// Write writes to the buffer.
+	Write(data []byte, offset int64) error
+
+	// Upload asynchronously uploads written data to GCS.
+	Upload()
+
+	// Status fetches the stat details of buffer.
+	Status()
```

Comment:
It should return something. Probably enumerate constants above ?

Reply:
This PR is just for buffer creation. This will be taken care of in a separate PR.